wagtail/contrib/frontend_cache/utils.py

```diff
@@ -132,36 +170,87 @@ def purge_urls_from_cache(urls, backend_settings=None, backends=None):
             backend.purge_batch(urls)
 
 
-def _get_page_cached_urls(page):
-    page_url = page.full_url
+def _get_page_cached_urls(page, cache_object=None):
```

Comment:
I don't think this should be an opaque "cache object" - the `get_full_url` implementation definitely heavily assumes it's a request.

Reply:
@RealOrangeOne Please see https://github.com/wagtail/wagtail/pull/12454 which solves that problem (this PR includes the changes from that branch)